src/globals.d

```diff
@@ -14,26 +14,50 @@ import ddmd.root.array;
 import ddmd.root.filename;
 import ddmd.root.outbuffer;
 
-template xversion(string s)
-{
-    enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
-}
-
 private string stripRight(string s)
 {
     while (s.length && (s[$ - 1] == ' ' || s[$ - 1] == '\n' || s[$ - 1] == '\r'))
         s = s[0 .. $ - 1];
     return s;
 }
 
-enum IN_GCC     = xversion!`IN_GCC`;
+immutable bool TARGET_LINUX, TARGET_OSX, TARGET_FREEBSD, TARGET_OPENBSD,
+               TARGET_SOLARIS, TARGET_WINDOS, TARGET_POSIX;
 
-enum TARGET_LINUX   = xversion!`linux`;
-enum TARGET_OSX     = xversion!`OSX`;
-enum TARGET_FREEBSD = xversion!`FreeBSD`;
-enum TARGET_OPENBSD = xversion!`OpenBSD`;
-enum TARGET_SOLARIS = xversion!`Solaris`;
-enum TARGET_WINDOS  = xversion!`Windows`;
+shared static this()
```

Comment:
That's the plan. :) I made these immutable runtime variables than enums because whenever dmd is turned into a cross-compiler, this `shared static this` can be easily modified to parse the cross-compilation command-line options at runtime and then set the target OS.

I should have mentioned that above, but wanted to make sure it passed the auto-tester first.